### DISCUSSION TITLE
NavigationView does not respect MVVM

### DISCUSSION THREAD

**Discussion by SuperJMN** [amwx/FluentAvalonia#470] [Q&A]
**Describe the bug**
In order to follow the MVVM pattern, you need to avoid having references to anything that is related to the underlying presentation technology in a ViewModel. This control makes it very difficult to follow that requirement, because for example, the Settings item is a UI control.

I've rather go with all models instead.

**Comment by amwx** [comment]
You can disable the built-in Settings item (property on NavView) and supply your own in the FooterMenuItems/FooterMenuItemsSource.

**Comment by SuperJMN** [comment]
That's a nice workaround for the settings case, but how to define templates? I haven't managed to bind my own model's name and icon to the NavigationViewMenuItems.

I've looked for a pure MVVM sample, but found nothing.
Thanks in advance.

**Comment by amwx** [comment]
I'm assuming you have different ViewModel types for different types of items? To handle that, use a DataTemplateSelector, which there is an example for on the NavigationView page:
https://github.com/amwx/FluentAvalonia/blob/4d847a8403b914d558baa79351a7866a02ec0b3e/samples/FAControlsGallery/Pages/FAControlsPages/NavigationViewPage.axaml#L55-L96

and the DataTemplateSelector class impl:
https://github.com/amwx/FluentAvalonia/blob/4d847a8403b914d558baa79351a7866a02ec0b3e/samples/FAControlsGallery/ViewModels/FAControlPages/NavViewPageViewModel.cs#L76-L87

(note in the sample app I put that class with the ViewModel, but that's just because I'm lazy - that can be put anywhere). This is a minimal version of the DataTemplateSelector. It could be made more extensible by storing a list of `IDataTemplate` which you could then declare an instance of your template selector as a resource with the the data templates desired and {StaticResource} it to the NavView. To handle the list, change the SelectTemplateCore method to loop over the templates to match its data type with the item argument (`template[i].Match(item)`).
